src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final int OPERATION_TYPE = 0;
+    private static final int OPERATION_TYPE_RESULT = 2;
+    private static final int OPERATION_TYPE_BUY = 1;
+    private static final int OPERATION_TYPE_SUPPLY = 0;
+    private static final int AMOUNT = 1;
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String OPERATION_RESULT = "result";
+    private static final String COMMA = ",";
+
+    public void getStatistic(String fromFileName, String toFileName) throws IOException {
+        String[] resultFromRead = readFromFile(fromFileName);
+        String data = infoGathering(resultFromRead);
+        writeToFile(toFileName, data);
+    }
+
+    private String[] readFromFile(String fromFileName)
```

Comment:
restructure your logic
read method should only read , return simple String, don't process data in any way. Transform your infoGathering into a method to analyze the data, or create a new private method